lib/component.js

```diff
@@ -327,6 +327,12 @@ class Component extends WebComponent {
     }
     this.initializing = true;
 
+    for (const attrsSchemaKey in this._attrsSchema) {
+      if (!Object.prototype.hasOwnProperty(this._attrs, attrsSchemaKey) && this._attrsSchema[attrsSchemaKey].required) {
+        this._logError(`Missing required attribute ${attrsSchemaKey} for ${this}`);
```

Comment:
Why not throw real Error ? and go kaboom

If you look at the type check in attrs, that throws a real error, just like other checks in panel.

https://github.com/mixpanel/panel/blob/43b1c4aa1ab48dbcc7d46c0c15ddf6cb4d32f2b1/lib/component.js#L573-L580

Reply:
My thought process was that this approach would allow all missing attributes to be logged, rather than blowing up on the first missing one and having the developer fix them one by one.
In reality, though I doubt this extra logging matters. The scenarios where this has caused me problems have been the addition of single properties. Direct throwing rather than logging and expecting the render to throw an exception is an indirect complexity with no real-world benefits.

Reply:
since this is connectedCallback (if the element is part of initial render, the whole dom tree of nested connectedCallbacks will fail with a proper stack trace). We only do try/catch log error in render because it's an asynchronous function call. Even there a render error is both logged and renderError event bubbled up (which could be sent to some error monitoring service e.g rollbar/sentry) 

 _logError doesn't do much. Just logs to console. So there is little way of knowing in the wild if some component is misconfigured.